Automattic/wp-calypso
# Stats: refactor to use notices hooks

Related to #77932 

## Proposed Changes

The PR refactored Stats Dashboard notice to leverage notices API rather than relying on notices in the initial state. It also removes redux related code not in use anymore.

This is a cleanup PR for [the Paid Stats Banner task](https://github.com/Automattic/wp-calypso/issues/78653).

## Testing Instructions

* Create a new JN site
* Connect the site to Jetpack
* Follow option 2 for Odyssey testing (sandbox widgets.wp.com)
* Ensure Opt Out Notice is shown and could be dismissed
* After 3~4 refreshes
* Ensure Feedback notice is shown and could be dismissed


![image](https://github.com/Automattic/wp-calypso/assets/1425433/1a8285d5-70cd-4067-a77b-44a4cead6c04)

![image](https://github.com/Automattic/wp-calypso/assets/1425433/27db4f59-21b1-4f6b-a98e-a59ebbba9431)



## Pre-merge Checklist
